src/components/ContentWidgets/WYSIWYG/WYSIWYGEditor.tsx

```diff
@@ -0,0 +1,25 @@
+import React from 'react';
+import { WidgetEditorProps } from '../../ContentMapping/ContentMapping';
+import { Editor } from 'react-draft-wysiwyg';
+import { EditorState, convertToRaw } from "draft-js";
+import '../../../../node_modules/react-draft-wysiwyg/dist/react-draft-wysiwyg.css';
+import './WYSIWYGEditor.css';
+
+export const WYSIWYGEditor: React.FC<WidgetEditorProps> = ({
+    editedContent,
+    setEditedContentOnChange
+}) => {
+    const contentState = editedContent.wysiwyg_state ?
+    JSON.parse(editedContent.wysiwyg_state) : convertToRaw(EditorState.createEmpty().getCurrentContent());
+
+    return <>
+        <Editor
+            initialContentState={contentState}
+            wrapperClassName="wysiwygWrapper"
+            editorClassName="wysiwygEditor"
+            // convert content to a raw JS structure, and save it as JSON string in database
+            onContentStateChange={(e) => 
+                setEditedContentOnChange("wysiwyg_state", JSON.stringify(e))}
```

Comment:
How much data does this use? What is the output for something simple with like 1 paragraph? What is the output for something complex that involves a ton of images and custom stuff and like 20 paragraphs?

Reply:
JSON:
"{\"blocks\":[{\"key\":\"3ps5\",\"text\":\"test2\",\"type\":\"unstyled\",\"depth\":0,\"inlineStyleRanges\":[{\"offset\":4,\"length\":1,\"style\":\"SUPERSCRIPT\"}],\"entityRanges\":[],\"data\":{}},{\"key\":\"3drk2\",\"text\":\"1\",\"type\":\"unordered-list-item\",\"depth\":0,\"inlineStyleRanges\":[],\"entityRanges\":[],\"data\":{}},{\"key\":\"6l793\",\"text\":\"2 \",\"type\":\"unordered-list-item\",\"depth\":0,\"inlineStyleRanges\":[],\"entityRanges\":[{\"offset\":1,\"length\":1,\"key\":0}],\"data\":{}},{\"key\":\"22qbo\",\"text\":\"\",\"type\":\"unstyled\",\"depth\":0,\"inlineStyleRanges\":[],\"entityRanges\":[],\"data\":{}},{\"key\":\"5k8tb\",\"text\":\" \",\"type\":\"atomic\",\"depth\":0,\"inlineStyleRanges\":[],\"entityRanges\":[{\"offset\":0,\"length\":1,\"key\":1}],\"data\":{}},{\"key\":\"af2cn\",\"text\":\"\",\"type\":\"unstyled\",\"depth\":0,\"inlineStyleRanges\":[],\"entityRanges\":[],\"data\":{}}],\"entityMap\":{\"0\":{\"type\":\"IMAGE\",\"mutability\":\"MUTABLE\",\"data\":{\"src\":\"data:image/jpeg;base64,/9j/4AAQSkZJRgABAQAAAQABAAD/2wCEAAkGBxMSEhUTExMWFhUXGBgaGRgXGBcXGhoYGBgYFxcYGBoYICggGhslGxgaIjEhJSkrLi4uGCAzODMtNygtLisBCgoKDg0OGhAQGC0lHR0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tK//AABEIAOEA4QMBIgACEQEDEQH/xAAcAAABBQEBAQAAAAAAAAAAAAAEAAECAwUGBwj/xABJEAABAgQEBAIHBQQHBwQDAAABAhEAAyExBBJBUQUiYXGBkQYTMqGx0fAHQsHh8RRScrIjM2JzgpKzFTRDU6LCw2Oj0tN0g5P/xAAZAQADAQEBAAAAAAAAAAAAAAAAAQIDBAX/xAAiEQEBAAICAwACAwEAAAAAAAAAAQIRAyESMUETUSJhcTL/2gAMAwEAAhEDEQA/APUZagC1m0oKfKpMIEg7l3/TakRI1blrdjcMQ1GiUxw9HULXqAfj+IjBRSFEqLjUN1Dbb0ixSS5oCQ9iz/nrtEUkaAs76d6N3LwquCRV70cD906gkbvaAHmF2NWfUENp4C1u8SDj2jVlO9iwF7b7Q0tItp0aoOoDxOWzO9x7no/huPhDB5QASSCWLHdtXeJSwXUTbTwv2rFaBcaAO9yw6ajR/dCU4CWJ9p931HgQbeW0BLAL1Z20saXN60hZcoJIq1co2oWGzaNDIFSAdj5sQRvt8omQ9nBZ/EmqYAiwoHqzd6X7xOUXari166v8IjMLOTZtnYHUN390SCAw8K3s2m14A4j0q4scxQk0TTxF/fSOblT3VzF9zC4rNMyc2hKlHwVX3xKRLDv1pHDn3e3o4SY4tKUCLGkEIxRF4FRMAimZOBERRO2wMUojl9+sFSOIq9k33PaMJGM0EavDMItY5jlT/wBR8IJ2VmhImkKcOXIjcwcxSgCEk6Gl4bDS5UtsqX6mp/KNWXPeNMcWO\",\"height\":\"400px\",\"width\":\"auto\"}},\"1\":{\"type\":\"IMAGE\",\"mutability\":\"MUTABLE\",\"data\":{\"src\":\"https://www.nationalgeographic.com/content/dam/news/2018/05/17/you-can-train-your-cat/02-cat-training-NationalGeographic_1484324.jpg\",\"height\":\"400px\",\"width\":\"auto\"}}}}"

HTML:
"<p>test<sup>2</sup></p>\n<ul>\n<li>l1</li>\n<li>l2</li>\n</ul>\n<p></p>\n<img src=\"https://www.nationalgeographic.com/content/dam/news/2018/05/17/you-can-train-your-cat/02-cat-training-NationalGeographic_1484324.jpg\" alt=\"undefined\" style=\"float:none;height: 400px;width: auto\"/>\n<p></p>\n"

Html uses much less data, but I had to do a lot of conversions like

```
convertToRaw(ContentState.createFromBlockArray(htmlToDraft(editedContent.wysiwyg_state).contentBlocks))
```